File: src/Chess.Api/Requests/MoveRequest.cs
namespace Chess.Api.Requests;

public record MoveRequest : Request
{
	public MoveRequest(SessionIdRequest sessionId, CellRequest from, CellRequest to)
	{
		this.SessionId = sessionId;
		this.From = from;
		this.To = to;
	}

	public override SessionIdRequest SessionId { get; }
	public CellRequest From { get; }
	public CellRequest To { get; }
}

File: src/Chess.Angular/ClientApp/src/app/Request/MoveRequest.ts
import { SessionIdRequest } from "./SessionIdRequest";
import { Request } from "./Request";
import { CellRequest } from "./CellRequest";

export class MoveRequest implements Request {
	constructor(public sessionId: SessionIdRequest, public from: CellRequest, public to: CellRequest) {
		//super(sessionId, "MoveRequest");
		
	}
	requestType: string = "MoveRequest";
	//new(sessionId: SessionIdRequest, from: CellRequest, to: CellRequest): MoveRequest;
	
	
	//sessionId: SessionIdRequest;
	// from: CellRequest;
	// to: CellRequest;
}
